contracts/cw2981-royalties/src/msg.rs

```diff
@@ -0,0 +1,138 @@
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+use cw721_base::msg::QueryMsg as CW721QueryMsg;
+
+#[derive(Serialize, Deserialize, Clone, PartialEq, JsonSchema, Debug)]
+#[serde(rename_all = "snake_case")]
+pub enum Cw2981QueryMsg {
+    // Should be called on sale to see if royalties are owed
+    // by the marketplace selling the NFT.
+    // See https://eips.ethereum.org/EIPS/eip-2981
+    RoyaltyInfo {
+        token_id: String,
+        // the denom of this sale must also be the denom returned by RoyaltiesInfoResponse
+        // this was originally implemented as a Coin
+        // however that would mean you couldn't buy using CW20s
+        // as CW20 is just mapping of addr -> balance
+        sale_price: u128,
```

Comment:
Should this be a `Uint128`? Might be nicer.

Reply:
Yep, good point